packages/react-core/src/helpers/util.ts

```diff
@@ -273,3 +273,79 @@ export const toCamel = (s: string) => s.replace(/([-_][a-z])/gi, camelize);
  * Copied from exenv
  */
 export const canUseDOM = !!(typeof window !== 'undefined' && window.document && window.document.createElement);
+
+/**
+ * This function is a helper for truncating text content on the left, leaving the right side of the content in view
+ *
+ * @param {any} row The text content to be truncated
+ */
+
+export const trimLeft = (row: any) => {
+  const trimContents = (row: { scrollWidth: any; offsetWidth: any }, node: ChildNode) => {
+    while (row.scrollWidth > row.offsetWidth) {
+      const childNode = node.firstChild;
+      const parent = node;
+      let parentVal = (node as HTMLInputElement).value;
+      // console.log("node", node);
+      // console.log('node value', parentVal);
+      // console.log("document.TEXT_NODE should be 3: ", document.TEXT_NODE);
+      // console.log("childNode", childNode)
+
+      const truncate = (input: string) => (input.length > 5 ? `${input.substring(0, 5)}...` : input);
+
+      if (!childNode) {
+        if (parentVal) {
+          let value = '...' + parentVal;
+          let sub = row.scrollWidth - row.offsetWidth;
+          let frac = row.offsetWidth / row.scrollWidth;
+          if (row.scrollWidth > row.offsetWidth) {
+
+
+            console.log("frac: ", frac);
+            console.log('sub', sub)
+            console.log("value length", value.length);
+            console.log("percent", frac * 100)
+            console.log("offsetwidth", row.offsetWidth)
+            console.log("scrollwidth", row.scrollWidth)
+            let newLength = frac;
+            value = '...' + value.substr(100);
+            parentVal = value;
+            console.log("parentVal", parentVal);
+            (node as HTMLInputElement).value = parentVal;
+            console.log((node as HTMLInputElement).value)
```

Comment:
Don't want to replace value with the truncated value ... should still be the value entered by the user. Is there a way to do this so that only the preview is truncated?